frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx

```diff
@@ -0,0 +1,171 @@
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const userInfo = useSelf()!;
+
+  const devPortfolios: DevPortfolio[] = [
+    {
+      name: 'test 1',
+      deadline: 222,
+      earliestValidDate: 111,
+      submissions: [],
+      uuid: 'xyz'
+    },
+    {
+      name: 'test 2',
+      deadline: 2,
+      earliestValidDate: 1,
+      submissions: [],
+      uuid: 'abc'
+    },
+    {
+      name: 'test abc',
+      deadline: 1,
+      earliestValidDate: 1,
+      submissions: [],
+      uuid: '123'
+    }
+  ];
+
+  const [devPortfolio, setDevPortfolio] = useState<DevPortfolio | undefined>(undefined);
+  // const [devPortfolios, setDevPortfolios] = useState<DevPortfolio[]>([]);
+  const [openPR, setOpenPR] = useState('');
+  const [reviewedPR, setReviewedPR] = useState('');
+
+  // useEffect(() => {
+  //   DevPortfolioAPI.getAllDevPortfolios().then((devPortfolios) => setDevPortfolios(devPortfolios));
+  // }, []);
+
+  const requestDevPortfolio = (
+    devPortfolioRequest: DevPortfolioSubmission,
+    devPortfolio: DevPortfolio
+  ) => {
+    devPortfolio?.submissions.push(devPortfolioRequest);
+    DevPortfolioAPI.requestDevPortfolio(devPortfolio).then((val) => {
+      if (val.error) {
+        Emitters.generalError.emit({
+          headerMsg: "Couldn't submit dev assignment!",
+          contentMsg: val.error
+        });
+      } else {
+        Emitters.generalSuccess.emit({
+          headerMsg: 'Dev Portfolio Assignment submitted!',
+          contentMsg: `The leads were notified of your submission and your submission will be graded soon!`
+        });
+      }
+    });
+  };
+
+  const submitDevPortfolio = () => {
+    if (!devPortfolio) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Dev Portfolio selected',
+        contentMsg: 'Please select a dev portfolio assignment!'
+      });
+    } else if (!openPR || !reviewedPR) {
+      Emitters.generalError.emit({
+        headerMsg: 'No opened or reviewed PR url submitted',
+        contentMsg: 'Please paste a link to a opened or reviewed PR!'
+      });
+    } else {
+      const newDevPortfolioSubmission: DevPortfolioSubmission = {
+        member: userInfo,
+        openedPRs: [openPR],
+        reviewedPRs: [reviewedPR],
+        status: 'pending'
+      };
+      requestDevPortfolio(newDevPortfolioSubmission, devPortfolio);
+      setDevPortfolio(undefined);
+      setOpenPR('');
+      setReviewedPR('');
+    }
+  };
+
+  return (
+    <div>
+      <Form className={styles.form_style}>
+        <h1>Submit Dev Portfolio Assignment</h1>
+        <p>Submit the recent pull requests you have opened or reviewed here.</p>
+        <div className={styles.inline}>
+          <label className={styles.bold}>
+            Select a Dev Portfolio Assignment <span className={styles.red_color}>*</span>
+          </label>
+          <div className={styles.center_and_flex}>
+            {devPortfolios && !devPortfolio ? (
```

Comment:
I don't think there needs to be `!devPortfolio`. It's probably better to still display the dropdown even with a dev portfolio selected